bigchaindb/lib.py

```diff
@@ -94,29 +95,23 @@ def write_transaction(self, transaction, mode):
 
     def _process_post_response(self, response, mode):
         logger.debug(response)
-        if response.get('error') is not None:
-            return (500, 'Internal error')
+
+        error = response.get('error')
+        if error:
+            return (500, error)
```

Comment:
Shouldn't we return an HTTP 400 "Bad Request" rather than 500 "Internal Server Error"? That's what [the HTTP API docs](http://docs.bigchaindb.com/projects/server/en/master/http-client-server-api.html#post--api-v1-transactions?mode=mode) say happens:

![screenshot from 2018-09-06 14-38-03](https://user-images.githubusercontent.com/1210951/45157861-95dfd680-b1e2-11e8-92dd-c34397b034a2.png)

If not, we need to update the HTTP API docs.

Reply:
This is how Tendermint docs explain the error field:

> only returns error if mempool.BroadcastTx errs (ie. problem with the app) or if we timeout waiting for tx to commit

This is thus a server error and it can occur even if the posted tx is valid.

Reply:
Ah, okay.